portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,76 @@
 
 package com.google.sps.servlets;
 
+import com.google.sps.servlets.DataServlet;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+/** Servlet that handles comments data */
+@WebServlet("/comment")
 public class DataServlet extends HttpServlet {
+    public static String NAME = "Comment";
+    public static String CONTNET = "content";
+    public static String TIMESTAMP = "timestamp";
+    public static String EMAIL = "email";
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Sue!</h1>");
+    Query query = new Query(NAME);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<Comment> new_comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+        long new_id = entity.getKey().getId();
+        String new_content = (String) entity.getProperty(CONTNET);
+        long new_timestamp = (long) entity.getProperty(TIMESTAMP);
+        String new_email = (String) entity.getProperty(EMAIL);
+        Comment new_comment = new Comment(new_id, new_content, new_timestamp, new_email);
+
+        new_comments.add(new_comment);
+    }
+
+    response.setContentType("application/json;"); // Send the JSON as the response
+    String json = new Gson().toJson(new_comments);
+    response.getWriter().println(json);
   }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Post user comment if a comment is submitted
```

Comment:
Same here, missing dot at the end. Same for the comments below.

Reply:
The dot is added at the end!